### ISSUE TITLE
Explainability for Link Prediction

### ISSUE THREAD

**Issue by RexYing** [pyg-team/pytorch_geometric#5925]
### 🚀 The feature, motivation and pitch

From #5804 , we already have explanation setting for node classification and graph classification, through the `task_level` argument of the `Explainer` class. 

We will create an additional option to enable link prediction explanation setting, and provide a basic implementation of explaining link prediction through e.g. GNNExplainer, SubgraphX or other methods. 

The idea is that through the general interface of `explain.base.ExplainerAlgorithm` class, we can plug in different explainability algorithms to perform link prediction explanation. 

A basic form of explanation output can be specified through the `Explanation` class.

**Comment by avgupta456** [comment]
In #5967, ModelTaskLevel is an Enum accepting `node`, `graph`, and `edge`. Can we understand link prediction as a subset of edge regression, or should we create a separate option for `link_pred`? In the `edge` task level, the current implementation doesn't allow classifying/regressing an edge not in the graph, but negative link prediction explanations might be desired. Thoughts?

Additionally, should a separate issue be created for edge classification/regression explanation?

**Comment by rusty1s** [comment]
I think this should be supported if you pass in an `edge_label_index` as additional argument to the model. Would this resolve your concern?

**Comment by avgupta456** [comment]
That's true, I can incorporate an edge_label_index. That solves most of the problem I think.

Additionally, I was wondering if it is okay to combine "Link Prediction" and "Edge classification/regression" under the same ModelTaskLevel.edge category, by considering link prediction as edge regression. I was unsure if link prediction is more common and would be better having a separate label.

**Comment by avgupta456** [comment]
Can we close this now?

**Comment by rusty1s** [comment]
Yes :)
